### PR TITLE
Fix issue where Search API endpoint would panic due to empty AuthConfig

### PR DESCRIPTION
**Pull request by tiborvass**
Fixes the failing docker-py test

Signed-off-by: Tibor Vass tibor@docker.com

### CODE REVIEW

File: registry/session.go [hunk 1]
```diff
@@ -89,13 +89,16 @@ func (tr *authTransport) RoundTrip(orig *http.Request) (*http.Response, error) {
 	tr.mu.Unlock()
 
 	if tr.alwaysSetBasicAuth {
+		if tr.AuthConfig == nil {
```
Comment (tiborvass) [inline registry/session.go]: this is just to prevent a potential panic, it's for precaution

**Comment by duglin** [comment]
is there a testcase that shows this? Or is it a pain to create?

**Comment by calavera** [comment]
@duglin every test_Search case failing for docker-py in every PR :wink:

**Comment by tiborvass** [comment]
@duglin sorry yeah, i updated the description after opening the PR

**Comment by calavera** [comment]
LGTM

**Comment by duglin** [comment]
LGTM just waiting for partial-janky  :-)

**Pull request merged**